sympy/physics/units/tests/test_quantities.py

```diff
@@ -528,6 +528,20 @@ def test_issue_22819():
     assert Da.scale_factor == 1.66053906660000e-24
 
 
+def test_issue_20288():
+    from sympy import sympify
+    from sympy.core.numbers import E
+    from sympy.physics.units import energy
+    u = Quantity('u')
+    v = Quantity('v')
+    SI.set_quantity_dimension(u, energy)
+    SI.set_quantity_dimension(v, energy)
+    u.set_global_relative_scale_factor(1, energy)
+    v.set_global_relative_scale_factor(1, energy)
```

Comment:
The scale factor is supposed to be set relative to another quantity object, not to a dimension. `energy` is a `Dimension`.

Reply:
> `energy` is a `Dimension`.

Then would something like this suffice .
```
>>> u.set_global_relative_scale_factor(1, joule)
>>> v.set_global_relative_scale_factor(1, joule)
```
Joule , Calories etc. would be considered as quantities of energy right ?

Reply:
I guess this is correct , I make a commit as soon as I get a green flag !

Reply:
@Upabjojr I have made the change , would appreciate a review !

Reply:
Yes, `joule` is a quantity corresponding to the dimension of `energy` (in every unit/dimension system).

Reply:
Thanks, I have made the change so you can check it out .Does the pr look good to go then ?